src/basepaths.rs

```diff
@@ -333,6 +333,8 @@ mod tests {
     use pretty_assertions::assert_eq;
 
     use std::fs;
+    use std::fs::File;
+    use std::os::unix::fs::PermissionsExt;
```

Comment:
I'm guessing this is not portable?

Reply:
It'll work on macOS and Linux, I think, but definitely not Windows. You can guard these with `#[cfg(...)]` directives. I think there's some examples in the codebase of that already.

Reply:
I set it up to test this only on Windows. Initially I had the hook being set up on Windows but without the `chmod`. Then there was a weird error, which was (I think) Windows not knowing where to find `bash`. At that point I skipped Windows entirely as the basic premise was alread being tested in the non-Windows environments.